### PR TITLE
Introduce audio config with music volume

### LINKED ISSUE #519: Make music volume configurable

### PR DESCRIPTION
**Pull request by JackCrumpLeys**
Closes #519.

### CODE REVIEW

File: crates/audio/Cargo.toml [hunk 1]
```diff
@@ -14,6 +14,7 @@ categories.workspace = true
 [dependencies]
 # DE
 de_core.workspace = true
+de_conf.workspace = true
```
Comment (Indy2222) [inline crates/audio/Cargo.toml:17]: Please sort dependencies alphabetically.

File: crates/conf/src/conf.rs [hunk 2]
```diff
@@ -45,6 +45,15 @@ pub struct Camera {
     #[ensure(*rotation_sensitivity > 0., "`rotation_sensitivity` must be greater than 0.0.")]
     pub rotation_sensitivity: f32,
 }
+
+#[derive(Deserialize, Config, Debug, Clone)]
+pub struct AudioConf {
+    pub music_enabled: bool,
+
+    #[is_finite]
+    #[ensure(*music_volume >= 0., "`music_volume` must be greater than 0.0.")]
```
Comment (Indy2222) [inline crates/conf/src/conf.rs:54]: * I would enforce that it is strictly larger than `0.0`. Alternatively, get rid of `music_enabled` and interpret `0.0` as disabled.
* Ensure that it is `<= 1.`.

File: crates/conf/src/conf.rs [hunk 3]
```diff
@@ -154,8 +172,21 @@ impl MultiplayerConf {
     }
 }
 
+impl AudioConf {
+    /// Whether music is enabled.
```
Comment (Indy2222) [inline crates/conf/src/conf.rs:176]: Please remove this doc comments, they just repeat name of the methods.

**Review by Indy2222 — CHANGES_REQUESTED** [review]
I had only a few minor comments on the code.

One comment **unrelated** to this PR: Config properties are `pub` and thus can not just be accessed but also changed from anywhere in the code.

I think that properties should be accessible only via getter methods so it is not possible to change them. Once configuration changing is to be implemented, it should be done via newly introduced setters, because:

* we could make it more defensive and assert that the new value is valid,
* and we could add any custom code tracking what was changed or cascading / propagating changes to related values if necessary.

**Comment by Indy2222** [comment]
I forgot to mention it the last time: please update the docs/, there is a page dedicated to the configuration.

**Comment by Indy2222** [comment]
I created #537 so such changes get easier in the future.

**Comment by Indy2222** [comment]
Sorry, this has happen in the meantime #542. Should be easy to fix.

**Comment by JackCrumpLeys** [comment]
> Sorry, this has happen in the meantime #542. Should be easy to fix.

fixed now

File: docs/src/conf.md [hunk 4]
```diff
@@ -44,3 +44,6 @@ est as if all properties are missing.
     rotation_sensitivity` radians. It must be a positive finite number.
   * `scroll_inverted` (bool; default: `false`) – if `true`, mouse wheel and
     touchpad scrolling is inverted.
+* `audio` (object) – audio configuration.
+  * `music_volume` (f32; default: `1.0`) – sets the music volume. It must be a finite
+    number between `0.0` and `1.0`. If set to 0 music will not play.
```
Comment (Indy2222) [inline docs/src/conf.md:49]: Please add a new line at the end of file.

**Review by Indy2222 — CHANGES_REQUESTED** [review]

**Review by Indy2222 — APPROVED** [review]

**Review by Indy2222 — APPROVED** [review]

**Review by Indy2222 — APPROVED** [review]

**Pull request merged**